### ISSUE TITLE
Chrome Extension

### ISSUE THREAD

**Issue by yoasif** [mozilla-lockwise/lockwise-addon#350]
> It might seem counter intuitive, but creating a Lockwise extension for Chrome would IMO make the overall FF ecosystem more attractive.

From: https://www.reddit.com/r/firefox/comments/dir065/use_firefox_lockwise_in_chrome/

**Comment by Millichrome** [comment]
FWIW, I don't think people who use lockwise would mainly use it on Chrome if an extension for Chrome existed.

It could be a nice bonus for FF users and it would make it easier for people to get into the FF "ecosystem."

**Comment by changecourse** [comment]
Thank you all for the interest! It's certainly something we've been thinking about as a consideration for our road map in the midterm future :)

**Comment by AkazaDorian** [comment]
I would love to see that, too. As far as every browser maker today is keeping their passwords in their own platforms, a free, cross-platform, trustable and easy-to-use password manager would be the true game changer.

**Comment by tmikaeld** [comment]
I've used Lastpass for many years and have now been using Bitwarden for the past 3 years, being fully portable and cross-platform is a requirement for any password manager. And I think that if people build trust into using Lockwise, that trust will also make them try Firefox and switch away from Chrome.

**Comment by ielashi** [comment]
I second the comments here. I'm a big FF user and have been using 1Password to manage my credentials. Even though I use FF and not Chrome, I can't move to a password manager if I feel like it'll corner me into using a specific browser.

**Comment by andreicristianpetcu** [comment]
Chrome support would be awesome. Firefox needs to continue being the good guy and promote colaboration.
A strong Chrome Web Store presence will get more users to switch.

**Comment by hg42** [comment]
I am a long time Firefox user on Linux and Windows, which is based mostly on all those amazing and well working extensions, that add features I need daily (e.g. Tree Style Tab, ContextSearch-web-ext and a lot more). The features that make the most difference are all about UI and usage.
Unfortunately Mozilla made Firefox let's say at least 30% less attractive by underestimating the importance of all these extensions. E.g. a lot of beloved and very useful extensions like Locationbar2 or Search-WP are still impossible.

Back to the topic, In some cases I still need Chrome because some web sites work better or a web site needs a feature that's not existent in Firefox.
I guess this situation will not go away in the near future (e.g. because of the market share).

I appreciate the step to offer sync features for Android that work cross browser. Additionally, I would like to see the same approach for bookmarks.

However, because I still need Chrome, I still need something independent like Bitwarden and xBrowserSync.

It should be clear, that a cross-browser approach lowers the hurdle for switching Browsers.

So, if Firefox offers advantages, a switch to Firefox would be easier.

On the other side, I guess you shouldn't fear to ease a switch from Firefox to Chrome, because from my experience, everyone who isn't a hard-nosed Firefox fan already switched to Chrome, when Firefox switched to Web-Extensions.

I have seen this all around me, on both professional and private domains, trying to offer something another browser already has and cutting away most things we all loved before at the same time simply suggests a switch, many users felt a loss of trust.

Btw. Vivaldi is another strong competition. It combines Chrome's engine with UI extensibility. I already tried this and apart from non-technical reasons only some details made this less attractive. Tree Style Tab is still more mature (e.g. drag and drop to bookmarks etc.) and offers some things not found in Tree Tabs and my multiline bookmark bar (which was an old Firefox extension and I now need to implement it via userchrome, which seems to be on deprecation path too) cannot work in Vivaldi because the toolbar is built dynamically (however changing the UI via styles is easy and straight forward).
I wish Firefox would copy and enhance more from Vivaldi instead of running behind Chrome.

**Comment by diogomax** [comment]
> FWIW, I don't think people who use lockwise would mainly use it on Chrome if an extension for Chrome existed.
> 
> It could be a nice bonus for FF users and it would make it easier for people to get into the FF "ecosystem."

People will probably not use on Chrome, but on Chromium based browsers, like Brave. You can install and use any extension from the Chrome Store. A Lockwise extension would just be perfect for using Brave Browser.

**Comment by jebofponderworthy** [comment]
I use three different browsers very regularly.  Lockwise is a time-waster for me without solid support for all major browsers.  I'll use Enpass until and unless the situation changes.

**Comment by godwheel** [comment]
I agree with everything everyone has said. I understand why Firefox wouldn't want to offer this, but as a Firefox user, I would prefer to use a Firefox password manager, but have the flexibility to use it on Chrome in the rare cases I have to use it.

Make this happen!

**Comment by jackmawer** [comment]
I just wanted to leave a comment to boost the voices of everyone here.

I think that it's invaluable to have Lockwise as openly available as possible to all platforms that are worth the development time to support, and I think it would be a missed opportunity not to reach out to the untapped userbase of Chrome and its derivatives and begin to bring them into the Firefox ecosystem.

Having Lockwise compatible with chrome would remove yet another of the stumbling blocks potentially stopping users from migrating, and the easier the transition between browsers can be made, the better.

**Comment by KevinRoebert** [comment]
> I just wanted to leave a comment to boost the voices of everyone here.

Same

**Comment by mattcoz** [comment]
Not just Chrome, but would be nice to have it on Chrome OS as well. I've been considering getting a Chromebook and having access to my passwords would be a must.

**Comment by BaronKrause** [comment]
Its worth it even if only for Chromebooks as their restricted to just using Chrome as a browser. (its possible to use Android/Linux installed browsers on Chrome OS, but the experience is so much worse that the natively installed browser).

Right now if people have a chromebook they will likely decide to not use Lockwise since it would mean using a password manager that cant be used on one of the platforms they use when there are other options that work on everything.

**Comment by ravinsinghd** [comment]
I am using LastPass. if lockwise add other browser support will switch. Because all of the development I am doing in edge (chromium), normal browsing in firefox.

**Comment by tmontanaro** [comment]
If the addon for Chrome will not exposed, a webpage to access the list of password outside Firefox could be at least useful (e.g., if I lose my smartphone, I'm outside and I ask my friend to access the internet, how can I access my passwords?).

**Comment by jamessauve** [comment]
Password manager has to be cross browser, man.  I was all pumped to get this working for my wife and I to share common household accounts.  This is a show stopper for me.  Even for just my accounts, I use both Chrome and Firefox daily for different sites. I really hope this feature is coming soon.

**Comment by apastuszak** [comment]
People pay good money for Lastpass, Bitwarden, 1Password, etc.  Why not develop extensions for other major browsers to allow password and bookmark sync and charge a $10 or $20 a year premium to defray the cost of development.

**Comment by cedrickcooke** [comment]
Have a Chromebook. Firefox is a second class citizen on that platform. Would love to have a password manager across all devices.

**Comment by barfin** [comment]
this would be great if a chrome user wants to switch to firefox it would be much easier since they don't need to make a second account and their passwords will be synced easily

**Comment by AkazaRenn** [comment]
Just to say it would be better to come out ASAP because cross-platform users will need a password manager anyways, if Lockwise is not available, they would use something else. And once they have been using that for long enough, it would be very tough to call them back.

**Comment by jwestco2** [comment]
I was just thinkig the Firefox Lockwise would be a good solution to my problem, but then realised I could only use it for the firefox browser.
I have come across occasions when firefox won't do what I need it to do, so I will then switch to an alternative browser and then my passwords would not be available.
So it is not a solution after all.
Fix this and I will almost certainly use it.

**Comment by draugvar** [comment]
I agree with the most of the people here. In the meanwhile I'm using Bitwarden...

**Comment by itsdarrylnorris** [comment]
Any updates on this? Can we get a response from the maintainer if this is even in their roadmap or something that they are planning on building?

**Comment by mnoorenberghe** [comment]
This is on the roadmap but it's not just a matter of simple changes to this extension since it leveraged the built-in Firefox password manager storage and Sync engine… in Chrome or other browsers we would need to re-implement that stuff or use other libraries for it. We are focusing on making the experience in Firefox great first and then we will expand out.

**Comment by ronaldscott** [comment]
The lack of cross-browser functionality makes Lockwise worthless to me. It's such a great idea, but this attitude from @mnoorenberghe 

> We are focusing on making the experience in Firefox great first and then we will expand out.

is short-sighted. Mozilla has a well-documented history of ending projects that don't get very much traction. This will happen to your project without cross-browser support. This is an existential problem for you.

**Comment by andreicristianpetcu** [comment]
as a contributor to Mozilla projects comments like yours make me sad @ronaldscott 
It's super frustrating to contribute my spare time to cool projects and get this attitude from users.
It's totally ok to use Firefox as your main browser with a cross browser password manager like bitwarden. Please don't put presure on mozilla employees and contributors. It's not productive.

**Comment by apastuszak** [comment]
I do not intend to put pressure on Mozilla.  I'm using Bitwarden now and happily paying for it.  I would rather pay Mozilla.  Developing Lockwise into a cross platform and cross browser plugin is a potential revenue stream for Mozilla.  As a very happy Firefox user, I'll happily pay for addon services that fill a need for my online life.

**Comment by ronaldscott** [comment]
@andreicristianpetcu I'm sorry to have made you sad; that wasn't my intention. I love the idea of Lockwise and want to see it succeed. I truly appreciate your efforts. I literally donate to the Mozilla Foundation on a yearly basis because of the efforts of developers like you. I consider myself to be on "team Mozilla" and "team Lockwise," but I firmly believe Lockwise will wither and die if it is only usable on one sub-5% market share browser and two mobile OS. I don't want to see that happen. 

I believe the vast majority of Firefox users cannot use it as their sole browser. I certainly cannot. There are things that don't render properly there, because of poorly-written websites/webapps that were developed in Chrome by developers who either lacked the resources or inclination to do cross-browser testing. Then there are developers like myself who are actually trying to perform cross-browser testing and need our password store available in both platforms. Right now, if we use Lockwise, that means we have an amazing, first-class experience in Firefox, and then if we want to get a password into Chrome we have to alt-tab, open Logins and Passwords, manually search for the login, click a copy button, alt-tab back, click on the password field, and paste. Even worse, it means our passwords spend time in a poorly-secured, open-to-any-running-process clipboard mechanism.

Perhaps my use case is the minority and users like me are not the base you're addressing, and if so, that's fine. I can accept that and regretfully move on. But if I'm right, and few Firefox users can find a way to limit themselves to using just Firefox, then that means most of the base needs cross-browser support.

Please don't let my opinions discourage you as a developer, though. I'm really just making a point to whomever is doing product management for Lockwise. I believe cross-browser support is vital and that the use case is limited and the project remains a curiosity for most users without it.

**Comment by itsdarrylnorris** [comment]
> I believe cross-browser support is vital and that the use case is limited and the project remains a curiosity for most users without it.

Indeed

**Comment by apastuszak** [comment]
If you live in Firefox, then Lockwise offers a password experience far better than what comes "out of the box" with other browsers.  But as soon as you need cross browser support and family sharing, you need to go elsewhere.

TAKE MY MONEY MOZILLA.  :-)

**Comment by m4kz** [comment]
This is very much needed as for example Im not allowed to use other than Chrome at work, and it is a sad reality for many people.

**Comment by 17specs** [comment]
Can there please be a way to export username and passwords to chrome via CSV file, I've been looking and I cant seem to find a way transfer credentials from one Firefox to another.

**Comment by andreicristianpetcu** [comment]
@17specs Firefox Nightly has this https://github.com/mozilla-lockwise/lockwise-addon/issues/332#issuecomment-638980335

Not sure when it will land in stable.

**Comment by mnoorenberghe** [comment]
In the meantime doesn't the Chrome option to import from Firefox address some of the issues? IIRC Chrome used to automatically import history from Firefox in the background after turning it on, does it not do that for passwords too? I know that's only one-way import to Chrome which isn't our idea case but if you just need convenient access to use Lockwise logins I think Chrome can do that. 

I have thought about how it would be easy to implement this feature accessing Firefox logins on the same computer for read/use, the more complicated part is the FxA/Sync part which doesn't have a maintained standalone JS library.

**Comment by EvilSpark** [comment]
Any update on it? Is it that tough to port them over?

**Comment by mnoorenberghe** [comment]
> Is it that tough to port them over? especially when ive heard that both uses WebExtension api thats quite similar since firefox quantum

See my last comment. This extension was relying on Firefox-specific private extension APIs to deal with Sync and storage so that would all have to be done a different way in another browser.

**Comment by voxpelli** [comment]
Now that desktop Safari also uses WebExtension it could be 3 for 1 (Chrome, Edge, Safari).

**Comment by filips123** [comment]
Safari's WebExtension API is still very limited in comparison with Firefox or Chromium.

**Comment by kumarharsh** [comment]
Would really love a cross-browser thing, for all the reasons stated above.

But what I'd also like to see (perhaps prioritised over other browsers' extensions) is Web access to my passwords like Bitwarden, Lastpass. The reasons are:
1. It's a decent stop-gap solution for users of other browsers.
1. It allows me to login in incognito windows or on other people's computers (while offering support or doing some one-off work on their laptops).
1. It gives me an option to completely replace other password manager.

**Comment by aquaspy** [comment]
A chromium based extension would be very nice as well as a desktop application for linux windows and mac (can be a electron one)

**Comment by edmael** [comment]
+1, it would be really nice to have Lockwise support in Chrome-based browsers.

**Comment by EmeraldSnorlax** [comment]
Adding my voice to the crowd; using Firefox at home, but in other places (work/school), Chrome is the only option.
It would also put my mind at ease over vendor lock in with passwords

**Comment by lakhbawa** [comment]
I like firefox and love to use it as a web developer, but have to switch to chrome occasionally for testing purposes.Firefox lock would be great to make permanent switch to firefox

**Comment by Yohe-Am** [comment]
Yes please.

**Comment by riyadhrazzaq** [comment]
Using Bitwaredn, still have not switched to Lockwise only because of this issue.

**Comment by janopae** [comment]
In order to attract organisations, it's crucial to support all major platforms. Otherwise, employers might be confused or even opposed to the decision of using Firefox Lockwise as enterprise password manager. On the other hand, enterprises are the ones most likely to pay money for a supported premium version of a password manager.

I am convinced that the importance of a Lockwise version for Chromium based browsers regarding the success of this service must not be underrated.

**Comment by mikkolukas** [comment]
I use another password manager at the moment, but do consider switching.

I looked into Firefox Lockwise, but the missing option to use it in other browsers was an instant turn off for me.
Maybe another time.

**Comment by cargaona** [comment]
I would love to have this functionality.

**Comment by Luxter77** [comment]
This would make my life so much easier

**Comment by KibSquib48** [comment]
I used Edge and now I'm using Vivaldi, Firefox is alright but I really just wanna use lockwise and I don't want to switch my whole browser just to have my passwords synced between my phone and PC

**Comment by CaptainN** [comment]
I would love to use LockWise on Windows and macOS the way it works on Android - for ALL apps, not just browsers.

**Comment by JakobJingleheimer** [comment]
The Firefox-only limitation is becoming a deal-breaker. I use Firefox for personal browsing, but (honestly sorry to say) Firefox just doesn't cut the mustard for development (such a relatively small market-share and has different/buggier behaviour than Chromium, slower to roll out new APIs, less performant, etc). That coupled with Lockwise barely working on iOS, it's only useful ~10% of the time ☹️

I would really like to see this project succeed, but so far it seems determined to fail.

EDIT: I'm not here to troll. I'm saying consciously refusing the vast-majority of the market is a terrible idea. Firefox accounts for [3.77% of browser usage](https://gs.statcounter.com/browser-market-share). Whilst you focus on that 4%, competitors can, will, and are capturing the other 96%, and Lockwise is extremely unlikely to re-capture them. _And_ those competitors also supply Lockwise's space, so not only will you lose the 96%, you will lose much if not all of the 4% you potentially already have. Mozilla does great work; I want you to win.

**Comment by andreicristianpetcu** [comment]
I wanted to look into working on this but I constantly found more high value Mozilla contributions I can make.

I also need a cross browser password manager but since nobody has time to look at this I switched to Bitwarden. It's free software, works in all browsers/platforms so it's a pretty sweet workaround to this issue. I totally recommend it. https://bitwarden.com/

**Comment by ZNackasha** [comment]
last pass has just changed there free model and would very much like to change to lockwise but cant because of this. 

PS i would much rather pay mozilla then lastpass for premium content.

https://blog.lastpass.com/2021/02/changes-to-lastpass-free/

**Comment by MendyBerger** [comment]
Would it be easier to create a Chromium extension that doesn't autofill forms, all it does is give you a small popup when you click on the extension icon with the username/password of the current site you're on?

This would probably be enough for users that primarily use Firefox but need to use other browsers occasionally.

**Comment by KibSquib48** [comment]
> last pass has just changed there free model and would very much like to change to lockwise but cant because of this.
> 
> PS i would much rather pay mozilla then lastpass for premium content.
> 
> https://blog.lastpass.com/2021/02/changes-to-lastpass-free/

off topic but man that change is super scummy, I get they need money to stay afloat but I'm pretty sure most users use password managers to sync their passwords from their phone to their pc

**Comment by apastuszak** [comment]
> > last pass has just changed there free model and would very much like to change to lockwise but cant because of this.
> > PS i would much rather pay mozilla then lastpass for premium content.
> > https://blog.lastpass.com/2021/02/changes-to-lastpass-free/
> 
> off topic but man that change is super scummy, I get they need money to stay afloat but I'm pretty sure most users use password managers to sync their passwords from their phone to their pc

What's annoying is that before LogMeIn bought LastPass, they would charge $20.00 for all the premium features, which included the mobile app.  Mobile was not free.  Then LogMeIn bought them.  They raised the price, but made a lot of the features part of the free tier.  Then they raised the price again.  Now they're basically back to their original plans, but with a much higher price tag.

**Comment by 1Dragoon** [comment]
I need to occasionally use Chrome, and I'd pay for lockwise if it supported chrome. I've really been wanting to get off of lastpass. I used to have a paid subscription to lastpass but dropped it once they stopped supporting firefox for Android. Now I REALLY want to get off of lastpass and I'm hoping I will be able to move to lockwise, but I can't because of no Chrome support.

FWIW I'm otherwise all in on Mozilla, including becoming a Rust developer.

**Comment by pipdibble** [comment]
This would be a great feature for developers that have to use multiple browsers to check for compatibility.

**Comment by VectorWpl** [comment]
Many users will probably migrate from LastPass in the coming months. Pity that this did not get any traction that would push for the development of chrome extension. Now even M$ is entering the race and it is the other way around - they are missing Firefox addon. I hope this will be picked up sometime in the future but for now, it seems like bitwarden is being endorsed here as a better replacement.

**Comment by sergioad** [comment]
This is a must have, specially for those using Chromium like the Raspberry OS, now that Google decoupled sync from Chromium more and more people would need a good password manager, specially those who jump often between Ubuntu on their PC or pi and Raspberry OS

**Comment by kumarharsh** [comment]
At this point in time, I'm really not very hopeful about this issue getting the love it needs sooner. Microsoft launched their own password manager extension rolled into Microsoft Authenticator on mobile and as an extension for desktop browsers. Sadly that doesn't support Firefox, but it's so much more feasible than Firefox's stance of just restricting it to one browser.

That said, I've come to realize that perhaps it's best to use a dedicated password manager company like Bitwarden or 1Password because they act much faster on feedback and have more features.

**Comment by cadeyrn** [comment]
> because they act much faster on feedback

What do you expect? It's clearly stated in the README of this repository:

> We may use this repository in the future to bring Lockwise to other browsers via an extension but for now it is unmaintained.

**Comment by sergioad** [comment]
Yes, that is unfortunate, however I hope someday it gets developed, for the
meantime I will stick to Microsoft Authenticator

**Comment by Necriis** [comment]
Yes, over the last few years, Microsoft has at least learned that rather than trying to force people to come to their solutions, it's better to go where people are.
Maybe Mozilla is not ready/mature yet... 😟

**Comment by martininsulander** [comment]
I think this would be great. And in the long run important to work in as many environments as people use it in. Myself, I'm using keepassxc for that reason. I'm thinking, one way would be support for syncing with other password managers, instead of an app, supposed to work in all kind of environments.

Thanks for all good work, and don't listen too much to the bad comments, most of them don't seem to know what they talk about.

**Comment by vfmatzkin** [comment]
This would be awesome. Still waiting for this feature to switch to Firefox! Sometimes I have no choice and the lack of integration is uncomfortable.

**Comment by jtagcat** [comment]
### GitHub Etiquette

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue. (You are literally sending e-mails or equivalent to 56 persons!)
* Subscribe to receive notifications on status change and new comments.
